src/Functions/UTCTimestampTransform.cpp

```diff
@@ -0,0 +1,184 @@
+#include <Columns/ColumnConst.h>
+#include <Columns/ColumnsDateTime.h>
+#include <Columns/ColumnString.h>
+#include <Columns/ColumnsNumber.h>
+#include <Columns/IColumn.h>
+#include <Common/DateLUT.h>
+#include <Common/LocalDateTime.h>
+#include <Core/DecimalFunctions.h>
+#include <DataTypes/DataTypeDateTime.h>
+#include <DataTypes/DataTypeDateTime64.h>
+#include <DataTypes/DataTypeNullable.h>
+#include <Functions/FunctionFactory.h>
+#include <Functions/IFunction.h>
+
+namespace DB
+{
+namespace ErrorCodes
+{
+    extern const int ILLEGAL_TYPE_OF_ARGUMENT;
+    extern const int ILLEGAL_COLUMN;
+    extern const int NUMBER_OF_ARGUMENTS_DOESNT_MATCH;
+}
+
+namespace
+{
+    template <typename Name>
+    class UTCTimestampTransform : public IFunction
+    {
+    public:
+        static constexpr auto name = Name::name;
+
+        String getName() const override { return name; }
+
+        size_t getNumberOfArguments() const override { return 2; }
+
+        bool useDefaultImplementationForConstants() const override { return true; }
+
+        bool isSuitableForShortCircuitArgumentsExecution(const DataTypesWithConstInfo & /*arguments*/) const override { return true; }
+
+        static FunctionPtr create(ContextPtr) { return std::make_shared<UTCTimestampTransform>(); }
+
+        DataTypePtr getReturnTypeImpl(const DataTypes & arguments) const override
+        {
+            if (arguments.size() < 2)
+                throw Exception(ErrorCodes::NUMBER_OF_ARGUMENTS_DOESNT_MATCH, "Function {}'s arguments number must be 2.", name);
+
+            DataTypePtr first_arg_type = arguments[0];
+            DataTypePtr second_arg_type = arguments[1];
+            if (first_arg_type->isNullable())
+            {
+                const DataTypeNullable * null_type = static_cast<const DataTypeNullable *>(first_arg_type.get());
+                first_arg_type = null_type->getNestedType();
+            }
+            if (second_arg_type->isNullable())
+            {
+                const DataTypeNullable * null_type = static_cast<const DataTypeNullable *>(second_arg_type.get());
+                second_arg_type = null_type->getNestedType();
+            }
```

Comment:
```suggestion
            DataTypePtr first_arg_type = removeNullable(arguments[0]);
            DataTypePtr second_arg_type = removeNullable(arguments[1]);
```
And btw, do we really want to allow the second argument to be `Nullable`?